### PR TITLE
add 'L' after _OPENSSL_VERSION_PRE_RELEASE literals, fixes #8021

### LINKED ISSUE #8021: Type incompatibility of new macro OPENSSL_VERSION_NUMBER
For the 3.0.0 pre-release the macro OPENSSL_VERSION_NUMBER has been re-implemented, now using a constant calculation rather than a literal. Commit 5d609f22d2 removed the type cast to `long`, such that now it reads:
```
#  define OPENSSL_VERSION_NUMBER        \
          ( (OPENSSL_VERSION_MAJOR<<28)  \
            |(OPENSSL_VERSION_MINOR<<20) \
            |(OPENSSL_VERSION_PATCH<<4)  \
            |_OPENSSL_VERSION_PRE_RELEASE )
```
This provides backward compatibility regarding the value, but not regarding the type of the expression.
Note that formerly the type was (unsigned) `long`, for instance as in:
```
# define OPENSSL_VERSION_NUMBER  0x10102000L
````
while now the inferred type is just `int`.

This creates type warnings/errors in contexts like
```
printf("OpenSSL compile-time version is 0x%lx\n", OPENSSL_VERSION_NUMBER);
```

I was about to suggest (re-)adding a type cast to `unsigned long` to the macro quoted above, but apparently this would cause other problem as mentioned in #7851.

### PR DESCRIPTION
**Pull request by DDvO**
Fixes #8021

### CODE REVIEW

File: include/openssl/opensslv.h [hunk 1]
```diff
@@ -126,9 +126,9 @@ const char *OPENSSL_version_build_metadata(void);
 # if !OPENSSL_API_4
 /* Synthesize OPENSSL_VERSION_NUMBER with the layout 0xMNN00PPSL */
 #  ifdef OPENSSL_VERSION_PRE_RELEASE
-#   define _OPENSSL_VERSION_PRE_RELEASE 0x0
+#   define _OPENSSL_VERSION_PRE_RELEASE 0x0L
 #  else
-#   define _OPENSSL_VERSION_PRE_RELEASE 0xf
+#   define _OPENSSL_VERSION_PRE_RELEASE 0xf:
```
Comment (mattcaswell) [inline include/openssl/opensslv.h]: Err - did you mean "L" here?

Comment (DDvO) [reply include/openssl/opensslv.h]: Oops, some editing mistake; corrected.

**Comment by DDvO** [comment]
When I updated this branch yesterday to fix the typo mentioned above, AppVeyor got stuck.
Since apart from making further changes to this patch I didn't find a way of triggering a restart 
I simply updated the copyright year from 2018 to 2019 ;-)

**Comment by mattcaswell** [comment]
Pushed.

**Pull request closed without merging**